crates/oxc_minifier/src/peephole/minimize_conditions.rs

```diff
@@ -733,6 +749,91 @@ impl<'a> PeepholeOptimizations {
         None
     }
 
+    /// Modify `expr` if that has `target_expr` as a parent, and returns true if modified.
+    ///
+    /// For `target_expr` = `a`, `expr` = `a.b`, this function changes `expr` to `a?.b` and returns true.
+    fn inject_optional_chaining_if_matched(
+        target_expr: &Expression<'a>,
+        expr: &mut Expression<'a>,
+        ctx: Ctx<'a, '_>,
+    ) -> bool {
+        fn inject(target_expr: &Expression<'_>, expr: &mut Expression<'_>) -> bool {
+            match expr {
+                Expression::StaticMemberExpression(e) => {
+                    if e.object.content_eq(target_expr) {
+                        e.optional = true;
+                        return true;
+                    }
+                    if inject(target_expr, &mut e.object) {
+                        return true;
+                    }
+                }
+                Expression::ComputedMemberExpression(e) => {
+                    if e.object.content_eq(target_expr) {
+                        e.optional = true;
+                        return true;
+                    }
+                    if inject(target_expr, &mut e.object) {
+                        return true;
+                    }
+                }
+                Expression::CallExpression(e) => {
+                    if e.callee.content_eq(target_expr) {
+                        e.optional = true;
+                        return true;
+                    }
+                    if inject(target_expr, &mut e.callee) {
+                        return true;
+                    }
+                }
+                Expression::ChainExpression(e) => match &mut e.expression {
+                    ChainElement::StaticMemberExpression(e) => {
+                        if e.object.content_eq(target_expr) {
+                            e.optional = true;
+                            return true;
+                        }
+                        if inject(target_expr, &mut e.object) {
+                            return true;
+                        }
+                    }
+                    ChainElement::ComputedMemberExpression(e) => {
+                        if e.object.content_eq(target_expr) {
+                            e.optional = true;
+                            return true;
+                        }
+                        if inject(target_expr, &mut e.object) {
+                            return true;
+                        }
+                    }
+                    ChainElement::CallExpression(e) => {
+                        if e.callee.content_eq(target_expr) {
+                            e.optional = true;
+                            return true;
+                        }
+                        if inject(target_expr, &mut e.callee) {
+                            return true;
+                        }
+                    }
+                    _ => {}
+                },
```

Comment:
Is it possible to create `ChainElement::as_expression_mut`? If that's possible, this can be:
```suggestion
                Expression::ChainExpression(e) => {
                    return inject(target_expr, e.as_expression_mut())
                },
```

Reply:
This pattern is rare, let's keep the overly long code 😅